### DISCUSSION TITLE
How do I redirect to another domain name?

### DISCUSSION THREAD

**Discussion by kyrofa** [nginx/kubernetes-ingress#5576] [Q&A] [answered]
Hey there. I know I can use annotations to, say, redirect http to https. However, I need to redirect one domain to another. As it so happens I'm specifically wanting to redirect my www subdomain to my root domain, but I'm also curious how to do this more generally. Is there a way to do that without enabling snippets? For example, the k8s community nginx ingress controller has the annotation `nginx.ingress.kubernetes.io/from-to-www-redirect`. It also has more generic `nginx.ingress.kubernetes.io/permanent-redirect`.

Thanks for your help!

**Comment by jasonwilliams14** [comment, marked as answer]
@kyrofa Hello there! If you are using our CRD, you can do this quite easily with `action.redirect`:

https://docs.nginx.com/nginx-ingress-controller/configuration/virtualserver-and-virtualserverroute-resources/#actionredirect

if you are using Ingress resource, then yea, you would need to do use snippets or create a custom annotation.
HTH

**Reply by kyrofa** [reply]
Yes I am indeed using an Ingress. Okay, thank you, I'll continue using snippets.

**Comment by brianehlert** [comment]
That said.
@jasonwilliams14 would you mind sharing the different ways this could be done with NGINX?
_Such as redirect which is essentially send back to the client to connect somewhere else vs a proxy style connection forward._

**Reply by jasonwilliams14** [reply]
@brianehlert A simple example could be something like the following:

```
server {
    listen 80;

    server_name example.com;
    return 301 https://$host$request_uri;
}
```
There couple different things we can think about such as setting up the server blocks.
let me put a few more in depth detailed examples and post them in this thread.
